plugins/Microsoft.DocAsCode.Build.UniversalReference/ModelConverter/ModelConverter.cs

```diff
@@ -0,0 +1,281 @@
+                    .Select(Mapper.Map<ReferenceViewModel, ApiNames>))
+                {
+                    references[reference.Uid] = reference;
+                }
+            }
+
+            // map references of children to ApiBuildOutput
+            Mapper.Initialize(cfg =>
+            {
+                cfg.AddProfile(new ApiBuildOutputProfile(supportedLanguages));
+            });
+            Mapper.Configuration.AssertConfigurationIsValid();
+            var childUids = model.Items[0].Children.Concat(model.Items[0].ChildrenInDevLangs.SelectMany(kv => kv.Value)).Distinct();
+            var children = new Dictionary<string, ApiBuildOutput>();
+            if (model.References != null)
+            {
+                foreach (var reference in model.References
+                    .Where(r => !string.IsNullOrEmpty(r.Uid) && childUids.Contains(r.Uid))
+                    .Select(Mapper.Map<ReferenceViewModel, ApiBuildOutput>))
+                {
+                    children[reference.Uid] = reference;
+                }
+            }
+
+            // map items to ApiBuildOutput
+            var items = Mapper.Map<List<ItemViewModel>, List<ApiBuildOutput>>(model.Items);
+            var result = items[0];
+            foreach (var child in items.Skip(1).Where(r => !string.IsNullOrEmpty(r.Uid)))
+            {
+                children[child.Uid] = child;
+            }
+
+            // fill in children
+            if (model.Items[0].Children != null || model.Items[0].ChildrenInDevLangs.Count > 0)
+            {
+                result.Children = ToApiListInDevLangs(model.Items[0].Children, model.Items[0].ChildrenInDevLangs, supportedLanguages)
+                    ?.Select(pair =>
+                    {
+                        return new ApiLanguageValuePair<List<ApiBuildOutput>>
+                        {
+                            Language = pair.Language,
+                            Value = pair.Value.Select(item => ResolveApiBuildOutput(item, children)).ToList()
+                        };
+                    }).ToList();
+            }
+
+            return result;
+        }
+
+        public static List<ApiLanguageValuePair<ApiNames>> ToApiListInDevLangsResolvingApiNames(string defaultValue, SortedList<string, string> values, string[] supportedLanguages, IReadOnlyDictionary<string, ApiNames> references)
+        {
+            if (defaultValue == null || supportedLanguages == null || supportedLanguages.Length == 0)
+            {
+                return null;
+            }
+            return ToApiListInDevLangs(defaultValue, values, supportedLanguages)
+                ?.Select(pair =>
+                {
+                    return new ApiLanguageValuePair<ApiNames>
+                    {
+                        Language = pair.Language,
+                        Value = ResolveApiNames(pair.Value, supportedLanguages, references)
+                    };
+                }).ToList();
```

Comment:
`?.Select` in L104, It cannot be null after Select()?

---
In reply to: [114462769](https://github.com/dotnet/docfx/pull/1618#discussion_r114462769) [](ancestors = 114462769)